main.py

```diff
@@ -263,77 +268,94 @@ def _load_flows(self):
-
-                if installed_flow["command"] == "delete":
-                    # No strict match
-                    if match_flow(flow, version, stored_flow["flow"]):
-                        deleted_flows_idxs.add(i)
-
-                elif installed_flow_obj == stored_flow_obj:
-                    if stored_flow["command"] == installed_flow["command"]:
-                        log.debug("Data already stored.")
-                        return
-                    # Flow with inconsistency in "command" fields : Remove the
-                    # old instruction. This happens when there is a stored
-                    # instruction to install the flow, but the new instruction
-                    # is to remove it. In this case, the old instruction is
-                    # removed and the new one is stored.
-                    stored_flow["command"] = installed_flow.get("command")
+                # No strict match
+                if match_flow(flow_dict, version, stored_flow["flow"]):
                     deleted_flows_idxs.add(i)
-                    break
-
-            if deleted_flows_idxs:
-                stored_flows = [
-                    flow
-                    for i, flow in enumerate(stored_flows)
-                    if i not in deleted_flows_idxs
-                ]
-            if should_persist_flow:
-                stored_flows.append(installed_flow)
-            stored_flows_box[switch.id]["flow_list"] = stored_flows
+
+            if not deleted_flows_idxs:
+                continue
+
+            stored_flows = [
+                flow
+                for i, flow in enumerate(stored_flows)
+                if i not in deleted_flows_idxs
+            ]
+            has_deleted_any_flow = True
+
+            if stored_flows:
+                stored_flows_box[switch.id][cookie] = stored_flows
+            else:
+                stored_flows_box[switch.id].pop(cookie, None)
+
+        if has_deleted_any_flow:
+            stored_flows_box["id"] = "flow_persistence"
+            self.storehouse.save_flow(stored_flows_box)
+            del stored_flows_box["id"]
+            self.stored_flows = deepcopy(stored_flows_box)
+
+    # pylint: disable=fixme
+    def _add_flow_store(self, flow_dict, switch):
+        """Try to add a flow dict in the store."""
+        installed_flow = {}
+        installed_flow["flow"] = flow_dict
+        installed_flow["created_at"] = now().strftime("%Y-%m-%dT%H:%M:%S")
+
+        stored_flows_box = deepcopy(self.stored_flows)
+        cookie = int(flow_dict.get("cookie", 0))
+        if switch.id not in stored_flows_box:
+            stored_flows_box[switch.id] = OrderedDict()
+
+        # TODO handle issue 23 (overlapping FlowMod add)
+        if not stored_flows_box[switch.id].get(cookie):
+            stored_flows_box[switch.id][cookie] = [installed_flow]
+        else:
+            stored_flows_box[switch.id][cookie].append(installed_flow)
```

Comment:
That's a good idea, Italo, and nope the migration isn't included. I guesstimate that in one hour or so I could write a script for this and ship with the release notes (or somewhere that's more appropriate), do you still that we'll use or any other user will use this? If they will, it's worth it, otherwise the least effort route would be to give a heads up of the breaking change in the release notes, but this assumes they're not using in prod. Let me know.

Reply:
Yes, you are right. I agree that the least effort route would be better instead of anticipating this possible use case. Let's wait on someone to complain about it and then we use the script you mentioned.

Reply:
Right. Sounds good.